apps/demos/Demos/FileUploader/CustomDropzone/jQuery/index.js

```diff
@@ -8,7 +8,16 @@ $(() => {
     uploadUrl: 'https://js.devexpress.com/Demos/NetCore/FileUploader/Upload',
     visible: false,
     onDropZoneEnter(e) {
-      if (e.dropZoneElement.id === 'dropzone-external') { toggleDropZoneActive(e.dropZoneElement, true); }
+      if (e.dropZoneElement.id === 'dropzone-external') {
+        const items = e.event.originalEvent.dataTransfer.items;
+
+        const isSingleFileDragged = items.length === 1;
+        const isValidFileExtension = `.${items[0].type.replace(/^image\//, '')}`;
+
+        if (isSingleFileDragged && this.option('allowedFileExtensions').includes(isValidFileExtension)) {
```

Comment:
We can create additional variables to make the code clearer.
It's also better to use `e.component` to get an instance because this api is available in all frameworks
```javascript
const allowedFileExtensions = this.option('allowedFileExtensions');
const isValidFileExtension = allowedFileExtensions.includes(draggedFileExtension);
```

I also suggest u splitting variables to 2 blocks:

```javascript
->additional variables<-

->meaningful variables which will be used in `if`<-
```